mlflow/telemetry/client.py

```diff
@@ -44,6 +41,80 @@ def __init__(self):
 
         # consumer threads for sending records
         self._consumer_threads = []
+        self._is_config_fetched = False
+        self.config = None
+        self._fetch_config()
+
+    def _fetch_config(self):
+        def _fetch():
+            try:
+                self._get_config()
+                self._is_config_fetched = True
+                if self.config is None:
+                    self._is_stopped = True
+                    _set_telemetry_client(None)
+                else:
+                    # this should happen before the first send_batch call
+                    with self._batch_lock:
+                        if self._pending_records:
+                            self._drop_invalid_records()
+            except Exception:
+                self._is_config_fetched = True
+                self._is_stopped = True
+                _set_telemetry_client(None)
+
+        self._config_thread = threading.Thread(
+            target=_fetch,
+            name="GetTelemetryConfig",
+            daemon=True,
+        )
+        self._config_thread.start()
+
+    def _get_config(self):
+        """
+        Get the config for the given MLflow version.
+        """
+        mlflow_version = self.info["mlflow_version"]
+        if config_url := _get_config_url(mlflow_version):
+            try:
+                response = requests.get(config_url, timeout=1)
+                if response.status_code != 200:
+                    return
+                config = response.json()
+                if (
+                    config.get("mlflow_version") != mlflow_version
+                    or config.get("disable_telemetry") is True
+                    or config.get("ingestion_url") is None
+                ):
+                    return
+
+                if get_source_sdk().value in config.get("disable_sdks", []):
+                    return
+
+                if sys.platform in config.get("disable_os", []):
+                    return
+
+                rollout_percentage = config.get("rollout_percentage", 100)
+                if random.randint(0, 100) > rollout_percentage:
+                    return
```

Comment:
I think this is equivalent to a full rollout.

Reply:
For example, we can generate a UUID when mlflow is imported, and save it as a file. UUID is a 128-bit integer so it can be scaled to [0, 1]. If the file already exists, we should read it.

Reply:
So what you proposed is user-based rollout, but our data is per-session and the reason was that we don't want to store files on user's machine?

Reply:
I'm not sure if my approach is user-based but it's a way to make the rollout gradual.

Reply:
I think your approach and the current approach only differs on whether it's user-based or session-based.
Cuz for session, the current implementation also makes sure only the rollout_percentage sessions are enabled for telemetry -- though each user may have multiple sessions.